lms/djangoapps/class_dashboard/tests/test_dashboard_data.py

```diff
@@ -126,36 +126,60 @@ def test_get_problemset_grade_distrib(self):
                 sum_attempts += item[1]
             self.assertEquals(USER_COUNT, sum_attempts)
 
-    def test_get_d3_problem_grade_distrib(self):
-
-        d3_data = get_d3_problem_grade_distrib(self.course.id)
-        for data in d3_data:
-            for stack_data in data['data']:
+    @patch('class_dashboard.dashboard_data.Client')
+    def test_get_d3_problem_grade_distrib(self, mock_client):
+        mock_client.return_value.modules.return_value.grade_distribution.return_value = [
+            {
+                'grade': 1,
+                'max_grade': 1,
+                'count': USER_COUNT
+            }
+        ]
+
+        # pylint: disable=unused-variable
+        with patch('class_dashboard.dashboard_data.settings') as mock_settings:
+            d3_data = get_d3_problem_grade_distrib(self.course.id)
+            for data in d3_data:
+                for stack_data in data['data']:
+                    sum_values = 0
+                    for problem in stack_data['stackData']:
+                        sum_values += problem['value']
+                    self.assertEquals(USER_COUNT, sum_values)
+
+    @patch('class_dashboard.dashboard_data.Client')
+    def test_get_d3_sequential_open_distrib(self, mock_client):
+        mock_client.return_value.modules.return_value.sequential_open_distribution.return_value = [{'count': 0}]
```

Comment:
The test count that we're checking for is actually 11 at the moment (stored in the USER_COUNT variable).